volmdlr/wires.py

```diff
@@ -2972,9 +2996,20 @@ class ClosedPolygon2D(ClosedPolygonMixin, Contour2D):
     def __init__(self, points: List[volmdlr.Point2D], name: str = ''):
         self.points = points
         self._line_segments = None
+        self.index = 0
 
         Contour2D.__init__(self, self.line_segments, name)
 
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        if self.index >= len(self.points):
+            self.index = 0
+            raise StopIteration
+        else:
+            self.index += 1
+            return self.points[self.index-1]
```

Comment:
Because in the previous if it raises an error, you don't need to use an else statement here. You can delete the indentation of the block after it.